_posts/en/newsletters/2019-04-09-newsletter.md

```diff
@@ -0,0 +1,309 @@
+a few lines of additional code (view source on that page to see the
+function) that uses the position of the typo characters to emphasize
+them in red:
+
+![Screenshot of the bech32 interactive demo with the typo address above](/img/posts/2019-04-bech32-demo.png)
+
+There's a limit to how many errors the `check()` function can specifically identify.
+After that it can still tell that an address contains an error, but it
+can't identify where to look in the address for the error.  In that
+case, it'll still return the address as invalid but it won't return the
+position details:
+
+```text
+>> segwit_addr_ecc.check('bc1qw508z6qejxtdg4y5r4zarvary0c5yw7kv8f3t4', 'bc')
+error: "Invalid"
+pos: null
+```
+
+In the case where there are other problems with the address, the `error`
+field will be set to a more descriptive message that may or may not
+include a position of the error.  For example:
+
+```text
+>> segwit_addr_ecc.check('bc1zw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4yolo', 'bc')
+error: "Invalid character"
+pos: Array [ 43 ]
+```
+
+You can review the [source][bech32 errors] for a complete list of
+errors.
+
+Although we spent a lot of time looking at errors in this mini tutorial,
+we've hopefully shown how easy it is to provide nice interactive
+feedback to users entering bech32 addresses on a web-based platform.  We
+encourage you to play around with the [interactive demo][] to get an
+idea of what your users might see if you make use of this bech32 address
+feature.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].  Note: all merges described for Bitcoin Core and LND
+are to their master development branches; some may also be backported to
+their pending releases.*
+
+- [Bitcoin Core #15596][] updates the `sendmany` RPC to remove the
+  `minconf` parameter.  For outputs received to your wallet, this
+  parameter allowed you to specify how many confirmations they must have
+  before you attempted to spend them.  Now the wallet defaults are
+  always used.  Those defaults are not to spend outputs received from
+  other people until they are confirmed and to optionally allow spending
+  unconfirmed change outputs from yourself depending on the
+  `spendzeroconfchange` configuration setting.  This is the same way the
+  more commonly used `sendtoaddress` RPC has always worked.
+
+- [LND #2885][] changes how LND attempts to reconnect to all of its
+  peers when coming back online.  Previously it attempted to open
+  connections to all its persistent peers at once.  Now it spreads the
+  connections over a 30 second window to reduce peak memory usage by
+  about 20% and to also reduce the number of concurrent events later due
+  to the events occurring on a regular interval, such as pings.
```

Comment:
I think the repeated "concurrent **events** later due to the **events**"" could be phrased slightly better. How about something like:

> Now it spreads the connections over a 30 second window to reduce peak memory usage by about 20%. This also means that messages that are sent on a regular interval, such as pings, do not happen at the same time for all peers.